src/ngx_lua_resty_lmdb_module.c

```diff
@@ -139,13 +195,47 @@ static ngx_int_t ngx_lua_resty_lmdb_init_worker(ngx_cycle_t *cycle)
         return NGX_ERROR;
     }
 
+    if (lcf->key_data.data) {
+
+        //u_char *key = lcf->key_data.data;
+        MDB_val enckey;
+        //enckey.mv_data = &key[0];
+        enckey.mv_data = lcf->key_data.data;
+        enckey.mv_size = 32;
+        if(lcf->encryption_type == 1){
+            cipher = (EVP_CIPHER *)EVP_chacha20_poly1305();
+        }
+        else {
+            cipher = (EVP_CIPHER *)EVP_aes_256_gcm();
+        }
+        //cipher = (EVP_CIPHER *)EVP_aes_256_gcm();
+        int block_size = EVP_CIPHER_block_size(cipher);
+        rc = mdb_env_set_encrypt(lcf->env, lmcf_encfunc, &enckey, block_size);
+        if (rc != 0) {
+            ngx_log_error(NGX_LOG_CRIT, cycle->log, 0, "unable to set LMDB encryption key: %s", mdb_strerror(rc));
+            return NGX_ERROR;
+        }
+        lcf->key_data.data = NULL;
+    }
+
     rc = mdb_env_open(lcf->env, (const char *) lcf->env_path->name.data, 0, 0600);
     if (rc != 0) {
         ngx_log_error(NGX_LOG_CRIT, cycle->log, 0,
                       "unable to open LMDB environment: %s", mdb_strerror(rc));
         return NGX_ERROR;
     }
 
+    rc = mdb_reader_check(lcf->env, &dead);
```

Comment:
this changeset seems already in master. please rebase this branch to master to eliminate this change.